File: backend/RankReach/RankReach/src/Core/Models/SoldierRequest.cs
namespace RankReach.Core.Models;

public class SoldierRequestCreate
{
    public int SoldierId { get; set; }
    public int RequestId { get; set; }
    public string? Comment { get; set; }
}

public class SoldierRequestRead
{
    public int SoldierRequestId { get; set; }
    public int SoldierId { get; set; }
    public string FirstName { get; set; } = "";
    public string LastName  { get; set; } = "";
    public string RequestName { get; set; } = "";
    public byte RequestStatus { get; set; }
    public string StatusName { get; set; } = "";
    public DateTime RequestDate { get; set; }
    public string? Comment { get; set; }
    public int? AppointmentId { get; set; }
    public DateTime? AppointmentDate { get; set; }
    public string? AppointmentLocation { get; set; }
}

public class SoldierRequestUpdate
{
    public int SoldierRequestId { get; set; }
    public byte RequestStatus { get; set; }
    public int? AppointmentId { get; set; }
}

File: rank-reach/src/types/soldierRequest.ts
export interface SoldierRequestCreate {
  soldierId: number;
  requestId: number;
  comment?: string | null;
}

export interface SoldierRequestRead {
  soldierRequestId: number;
  soldierId: number;
  firstName: string;
  lastName: string;
  requestName: string;
  requestStatus: number; // status id
  statusName: string | null; // joined name (can be null)
  requestDate: string; // ISO string from API
  comment: string | null;

  appointmentId?: number | null;
  appointmentDate?: string | null; // ISO or null
  appointmentLocation?: string | null; // or null
}

export interface SoldierRequestUpdate {
  soldierRequestId: number;
  requestStatus: number; // 0=unanswered, 1=approved, 2=rejected
  appointmentId?: number | null;
}
